### DISCUSSION TITLE
Unattended plugins installation

### DISCUSSION THREAD

**Discussion by Pivert** [neoclide/coc.nvim#3524] [Q&A] [answered]
Hi,

I need to deploy coc & plugins with an unattended script because I'm building a container that will be used in an isolated network.

I did try:
- RUN npm install -g coc-spell-checker coc-prettier coc-git coc-pyright coc-json coc-docker coc-yaml
It installs the plugins, but it's not active in neovim. I need to re-issue a CocInstall which re-downloads the plugin.
- Same if using Yarn
- RUN nvim -es -i NONE -c "CocInstall coc-spell-checker coc-prettier coc-git coc-pyright coc-json coc-docker coc-yaml" -c "qa"
This last try just does nothing, I guess the 'qa' is issued too early, bafore CocInstall finished.

How should I run an unattended installation (in a docker container) ?

**Comment by chemzqm** [comment, marked as answer]
https://github.com/neoclide/coc.nvim/wiki/Using-coc-extensions#install-extensions

**Comment by Pivert** [comment]
OMG, how did I miss that in the documentation. Thanks !

**Comment by Pivert** [comment]
Hi,

This does not work any more.

```
$time vim -c 'CocInstall -sync coc-json coc-html|q'

real    0m0.206s
user    0m0.190s
sys     0m0.027s
09:48:45[Z690-nvimdocker-1598574][pivert] /workdir
$time vim -e -c 'CocInstall -sync coc-json coc-html|q'

real    0m0.222s
user    0m0.181s
sys     0m0.051s
09:48:50[Z690-nvimdocker-1598574][pivert] /workdir
$vim --version
NVIM v0.8.1
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by runner@fv-az178-366

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/share/nvim"

Run :checkhealth for more info
```

checkhealth is fine, and running the CocInstall from a vim session works. It seems like the -sync does not work (quits immediately).
